app/controllers/errors_controller.rb

```diff
@@ -38,15 +56,73 @@ def notify_subscribers
   end
 
   def resolve
-    if !@error.resolved_at.nil?
-      @error.update_attributes(resolved_at: nil)
+    #has values when we resolve issues through the error sidebar
+    errors_per_page = params[:per_page] || 5
+    page = params[:page]
+    if params[:error_ids]
+      ids = params[:error_ids]
+      resolved = to_boolean(params[:resolved]) if params[:resolved]
+      resolve_issues(ids, resolved, errors_per_page, page, @error)
+    elsif params[:individual_resolve]
+      ids = [@error.id]
+      resolved = !@error.resolved_at.nil?
+      resolve_issues(ids, resolved, errors_per_page, page, @error)
     else
-      @error.update_attributes(resolved_at: DateTime.now)
+      raise 'Could not find error!'
     end
   end
 
   private
 
+  def resolve_issues (ids, resolved = true, errors_per_page, page, current_error)
+    errors = current_error.website.grouped_issues.order('last_seen DESC')
+    if resolved
+      GroupedIssue.where(id: ids).update_all(resolved_at: nil, status: 3) # 3 = unresolved
+      errors = errors.where('resolved_at IS NOT NULL')
+    else
+      GroupedIssue.where(id: ids).update_all(resolved_at: DateTime.now, status: 2) # 2 = resolved
+      errors = errors.where(resolved_at: nil).page(page)
+    end
+    @sidebar = errors.page(page).per(ids.size).offset(errors_per_page)
+    @pagination = errors.page(page).per(errors_per_page).offset(ids.size)
+  end
+
+  def to_boolean(str)
+    return true if str=="true"
+    return false if str=="false"
+    return nil
+  end
+
+  def aggregations (issues)
```

Comment:
we should move this method in grouped_issue.rb model file and notice the space between the name of the method and argument, remove that.